lib/src/common/js_proxy.dart

```diff
@@ -145,11 +145,32 @@ dynamic dartValue(jsValue) {
       }
       _addDartInstance(jsValue, dartMap);
       return dartMap;
+    } else if (jsValue is JsFunction) {
+      return _dartType(jsValue);
```

Comment:
This if should probably be moved out one more level. Also, this should probably only return if _dartType(jsValue) is not null.

I would also add a TODO to figure out what we should do with general js functions. We could wrap them in a dart closure which invokes them possibly? It would have to just take one argument which is a list of all arguments to pass to the js function, but maybe that is ok? @sigmundch what do you think we should do here?